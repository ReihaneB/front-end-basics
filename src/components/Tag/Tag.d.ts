import React from 'react';

interface TagProps {
  children: React.ReactNode;
}
