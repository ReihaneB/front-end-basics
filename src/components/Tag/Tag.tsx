import { memo } from 'react';

import type { TagProps } from './Tag.d';

function Tag({
  children,
  ...rest
}: TagProps) {
  return (
    <div
      data-testid="tag-component"
      tabIndex={-1}
      {...rest}
    >
      {children}
    </div>
  );
}

export default memo(Tag);
