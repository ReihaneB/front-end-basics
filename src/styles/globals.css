:root {
  /* colors */
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  border: 0;
  font: inherit;
  vertical-align: baseline;
}

html,
body {
  max-width: 100vw;
  overflow-x: hidden;
}

details,
figcaption,
figure,
footer,
hgroup,
menu,
nav {
  display: 'block',
}

ol,
ul {
  list-style: none;
}

blockquote,
q {
  quotes: none;
}

blockquote::before,
blockquote::after,
q::before,
q::after {
  content: '';
  content: none;
}

table {
  border-collapse: collapse;
  border-spacing: 0;
}

a {
  color: inherit;
  text-decoration: none;
}

h1 {
  font-size: 30px;
  line-height: 38px;
  font-weight: bold;
}

h2 {
  font-size: 20px;
  line-height: 24px;
  font-weight: bold;
}

h3 {
  font-size: 16px;
  line-height: 20px;
  font-weight: bold;
}

h4 {
  font-size: 14px;
  line-height: 18px;
}

h5 {
  font-size: 12px;
  line-height: 16px;
}

p {
  font-size: 16px;
  line-height: 24px;
}