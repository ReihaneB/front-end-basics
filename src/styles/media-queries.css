@custom-media --small-mobile (max-width: 375px);
@custom-media --mobile (max-width: 767px);
@custom-media --small-tablet (max-width: 957px);
@custom-media --tablet (max-width: 1199px);
@custom-media --ultra-wide (min-width: 1920px);