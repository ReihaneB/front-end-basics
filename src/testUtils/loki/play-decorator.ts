/* eslint-disable @typescript-eslint/ban-ts-comment */
// @ts-expect-error
import createAsyncCallback from '@loki/create-async-callback';
// @ts-expect-error
import isLokiRunning from '@loki/is-loki-running';
import type {PlayFunction} from '@storybook/csf';
// @ts-expect-error
import type {ReactFramework} from '@storybook/react';

async function waitForDocumentLoaded(): Promise<void> {
  if (document.readyState === 'loading') {
    return new Promise(resolve => {
      document.addEventListener('DOMContentLoaded', () => resolve());
    });
  }

  return Promise.resolve();
}

function enablePointerEvents(): void {
  const styleElement = document.createElement('style');
  document.documentElement.appendChild(styleElement);
  styleElement.sheet?.insertRule('* {pointer-events: auto !important}');
}

export function lokiPlayDecorator(
  target: PlayFunction<ReactFramework>
): PlayFunction<ReactFramework> {
  return async context => {
    const lokiRunning = isLokiRunning();
    const asyncCallback = createAsyncCallback();

    if (lokiRunning) {
      await waitForDocumentLoaded();
      enablePointerEvents();
    }

    try {
      await target(context);
    } finally {
      asyncCallback();
    }
  };
}
