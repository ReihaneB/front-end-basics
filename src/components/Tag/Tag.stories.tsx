import type { Meta, StoryObj } from '@storybook/react';
import { expect, userEvent, waitFor, within } from '@storybook/test';

import { lokiPlayDecorator } from '@/testUtils/loki/play-decorator';

import Tag from './Tag';

import styles from './Tag.stories.module.css';

const meta: Meta<typeof Tag> = {
  title: 'Components/Tag',
  component: Tag,
};

export default meta;
type Story = StoryObj<typeof Tag>;

const defaultArgs = {
  children: 'Tag content',
};

export const Playground: Story = {
  args: {
    ...defaultArgs,
  },
};

export function LongText() {
  return (
    <div className={styles.container}>
      <Tag>
        This is a very long text that should be going to the next line
      </Tag>
    </div>
  );
}

LongText.storyName = 'Long text';

export const NotFocusTab: Story = {
  args: {
    ...defaultArgs,
  },
  play: lokiPlayDecorator(
    async ({ canvasElement }) => {
      const canvas = within(canvasElement);

      await userEvent.tab();

      await waitFor(() => expect(canvas.getByTestId('tag-component')).not.toHaveFocus());
    }
  ),
};
