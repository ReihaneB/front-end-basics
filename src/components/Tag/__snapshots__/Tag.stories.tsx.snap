// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Tag LongText smoke-test 1`] = `
<main class="inter-normal">
  <div class="src-components-Tag-Tag-stories-module__container--aL11d">
    <div data-testid="tag-component"
         tabindex="-1"
    >
      This is a very long text that should be going to the next line
    </div>
  </div>
</main>
`;

exports[`Components/Tag NotFocusTab play-test 1`] = `
<main class="inter-normal">
  <div data-testid="tag-component"
       tabindex="-1"
  >
    Tag content
  </div>
</main>
`;

exports[`Components/Tag Playground smoke-test 1`] = `
<main class="inter-normal">
  <div data-testid="tag-component"
       tabindex="-1"
  >
    Tag content
  </div>
</main>
`;
